src/main/java/crazypants/util/ClientUtil.java

```diff
@@ -47,18 +51,34 @@ public static void doGasLevelUpdate(int x, int y, int z, PacketGasLevel pkt) {
         con.readFromNBT(pkt.tc, TileConduitBundle.NBT_VERSION);
     }
 
-    public static void spawnFarmParcticles(Random rand, BlockCoord bc) {
-        double xOff = 0.5 + (rand.nextDouble() - 0.5) * 1.1;
-        double yOff = 0.5 + (rand.nextDouble() - 0.5) * 0.2;
-        double zOff = 0.5 + (rand.nextDouble() - 0.5) * 1.1;
-        Minecraft.getMinecraft().theWorld.spawnParticle(
-                "portal",
-                bc.x + xOff,
-                bc.y + yOff,
-                bc.z + zOff,
-                (rand.nextDouble() - 0.5) * 1.5,
-                -rand.nextDouble(),
-                (rand.nextDouble() - 0.5) * 1.5);
+    public static void spawnFarmParticles(Random rand, List<BlockCoord> coords) {
+        // 0 = All, 1 = Decreased, 2 = Minimal
+        int particleSetting = Minecraft.getMinecraft().gameSettings.particleSetting;
+        if (particleSetting >= 2 /* Minimal */) {
+            return;
+        }
+
+        int particles = particleSetting == 0 ? 6 /* All */ : 3 /* Decreased */;
+
+        World world = Minecraft.getMinecraft().theWorld;
+        EffectRenderer effectRenderer = Minecraft.getMinecraft().effectRenderer;
```

Comment:
Any reason it was changed from spawnParticle?

Reply:
spawnParticle uses random 30% reduction for number of particles
i want more control over this number

it also contains terrible impl of particle registry (when particle name compared one by one with all possible variants)

Reply:
Good catch. Just checked it and it was a little much.
Have you tested this in game, sp and server (runServer)?

Reply:
> Good catch. Just checked it and it was a little much. Have you tested this in game, sp and server (runServer)?

tested in sp and server